src/js/model/Frame.js

```diff
@@ -0,0 +1,47 @@
+function Frame() {
+  this._frame = [];
+}
+
+
+Frame.prototype.getFrame = function() {
+  return this._frame;
+};
+
+Frame.prototype.addRoll = function(roll) {
+  this.isWrong(roll)
+  this._frame.push(roll);
+};
+
+Frame.prototype.sumRolls = function() {
+  return (this._frame[0] + this._frame[1]);
+}
+
+Frame.prototype.isNew = function() {
+  return (this.getFrame().length === 1);
+};
+
+Frame.prototype.isFull = function() {
+  return (this.getFrame().length === 2);
+};
+
+Frame.prototype.isStrike = function() {
+  return (this._frame[0] === 10);
+};
+
+Frame.prototype.isSpare = function() {
+  if (!this.isStrike()) {
+    return (this.sumRolls() === 10);
```

Comment:
The brackets here in the `return` are interesting — were you worried about precedence? I.e. were you trying to avoid `(return this.sumRolls()) === 10`? Or something else?